processing/src/main/java/io/druid/segment/DimensionHandlerUtils.java

```diff
@@ -172,10 +185,56 @@ private static ColumnValueSelector getColumnValueSelectorFromDimensionSpec(
   private static <ColumnSelectorStrategyClass extends ColumnSelectorStrategy> ColumnSelectorStrategyClass makeStrategy(
       ColumnSelectorStrategyFactory<ColumnSelectorStrategyClass> strategyFactory,
       String dimName,
-      ColumnCapabilities capabilities
+      ColumnCapabilities capabilities,
+      boolean hasExFn
   )
   {
-    capabilities = getEffectiveCapabilities(dimName, capabilities);
+    capabilities = getEffectiveCapabilities(dimName, capabilities, hasExFn);
     return strategyFactory.makeColumnSelectorStrategy(capabilities);
   }
+
+  private static final Pattern LONG_PAT = Pattern.compile("[-|+]?\\d+");
+
+  // TopNBinaryFn relies on this returning the same Long object when no conversion is needed
+  public static Long convertObjectToLong(Object valObj)
+  {
+    if (valObj instanceof Long) {
+      return (Long) valObj;
+    } else if (valObj instanceof Number) {
+      return ((Number) valObj).longValue();
+    } else if (valObj instanceof String) {
+      try {
+        String s = ((String) valObj).replace(",", "");
+        return LONG_PAT.matcher(s).matches() ? Long.valueOf(s) : Double.valueOf(s).longValue();
+      }
+      catch (Exception e) {
+        throw new ParseException(e, "Unable to parse value[%s] as long", valObj);
+      }
+    } else {
+      throw new ParseException("Unknown type[%s]", valObj.getClass());
+    }
+  }
+
+  // TopNBinaryFn relies on this returning the same Float object when no conversion is needed
+  public static Float convertObjectToFloat(Object valObj)
+  {
+    if (valObj == null) {
+      return 0.0f;
+    }
+
+    if (valObj instanceof Float) {
+      return (Float) valObj;
+    } else if (valObj instanceof Number) {
+      return ((Number) valObj).floatValue();
+    } else if (valObj instanceof String) {
+      try {
+        return Float.valueOf(((String) valObj).replace(",", ""));
```

Comment:
I can imagine it could become a source or bugs, because in some Locales comma is used as a separator between integer and fractional parts. So removing commas from `"1,0"` will silently produce `10.0f` instead of failing with exception.